comps/cores/mega/gateway.py

```diff
@@ -622,31 +622,41 @@ def __init__(self, megaservice, host="0.0.0.0", port=8889):
             host,
             port,
             str(MegaServiceEndpoint.RETRIEVALTOOL),
-            Union[TextDoc, EmbeddingRequest, ChatCompletionRequest],  # ChatCompletionRequest,
-            Union[RerankedDoc, LLMParamsDoc],  # ChatCompletionResponse
+            ChatCompletionRequest,
```

Comment:
Are we not expecting TextDoc / EmbeddingRequest in future retrieval request? @ftian1 @lvliang-intel

Reply:
Good catch. I think we should also support TextDoc and EmbeddingRequest inputs. Let me add them back in.

Reply:
@xuechendi  I added the other input types back. Please review.